tests/support/server.tcl

```diff
@@ -302,13 +302,20 @@ proc run_external_server_test {code overrides} {
 
     if {[llength $::servers] == 1} {
         r flushall
+    } else {
+        puts "Warning: nested start_server statements in external server mode!"
     }
 
     # store overrides
     set saved_config {}
     foreach {param val} $overrides {
         dict set saved_config $param [lindex [r config get $param] 1]
         r config set $param $val
+
+        # If we enable appendonly, wait for for rewrite to complete
+        if {$param == "appendonly" && $val == "yes"} {
+            waitForBgrewriteaof r
+        }
```

Comment:
can you explain this?
- which test was it needed for and why?
- i'd imagine that tests that declare they wanna enable AOF cannot run in external mode.
- i'd imagine that some external servers won't support waitForBgrewriteaof (depends on missing info fields)

Reply:
Tests that don't assume they can inspect the AOF file but only do `debug loadaof` (assuming `needs:debug` is allowed) can run in external mode. There is also some logic around `start_server` to actually configure the server as needed, the problem is that if a test begins with `bgsave` (or some other `bg` operation) right after we did `config set appendonly` it will fail - the purpose of this wait is to handle that.

Added a comment as well.

Reply:
two random thoughts that we can dismiss.
1. we can put that wait with some condition only to run if the old value is different than `yes`. but on the other hand, this wait is harmless (if there's no BG process in progress, it'll exit immediately).
2. which leads me to the second thought, maybe we want this wait (and also wait for other BG processes) unconditionally, i.e. regadless of the configuration that's applied.
the the difference between and external and a spawned up server, is that we don't know it's state, so we do `flushdb`, and maybe we should also wait till it doesn't have any BG process, unrelated to the config we apply.